Allow PHP8.3

heiglandreas opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

This allows usage of the package using PHP8.3

heiglandreas committed:
Allow PHP8.3

Signed-off-by: Andreas Heigl <andreas@heigl.org>

Ocramius commented:
Lock file needs a hash refresh

heiglandreas commented:
That's the least of all issues...

> 
> Reading package lists...
> Building dependency tree...
> Reading state information...
> E: Unable to locate package php8.3-apcu
> E: Couldn't find any package by glob 'php8.3-apcu'
> E: Unable to locate package php8.3-memcached
> E: Couldn't find any package by glob 'php8.3-memcached'
> E: Unable to locate package php8.3-mongodb
> E: Couldn't find any package by glob 'php8.3-mongodb'
> E: Unable to locate package php8.3-redis
> E: Couldn't find any package by glob 'php8.3-redis'

https://github.com/laminas/laminas-diagnostics/actions/runs/6954575541/job/18921670184?pr=74

Ocramius commented:
Ah yes, I think we solved that recently for laminas-session?

froschdesign commented:
See: https://github.com/laminas/laminas-session/blob/e733d986b30704831c6ec530d06688faf6f0856a/.laminas-ci/install-mongodb-extension-via-pecl.sh

heiglandreas committed:
Allo platform requirements to be ignored on 8.3

At least for now...

Signed-off-by: Andreas Heigl <andreas@heigl.org>

heiglandreas committed:
Add installation instructions for pecl-extensions

Otherwise those extensions would not be installed correctly on PHP8.3

Signed-off-by: Andreas Heigl <andreas@heigl.org>

heiglandreas committed:
Handle PHP8.3 parse-error of ini-files

PHP8.3 now emits a syntax error warning in addition to returning
`false` now.
Previously the error was silently ignored and the returned value was an
empty array.

The `@` silcences the warning. As this is done during the check
whether the ini-file can be parsed or not it should be safe here to not
output the content of the warning.

Signed-off-by: Andreas Heigl <andreas@heigl.org>

heiglandreas commented:
Stupid question perhaps: Why are we going through all the hassle of setting up Memcached and Redis and Mongo et all to then just ignore those tests? 

```
There were 10 skipped tests:

1) LaminasTest\Diagnostics\ChecksTest::testRabbitMQ
RabbitMQ tests are not enabled; enable them in phpunit.xml

/github/workspace/test/ChecksTest.php:100

2) LaminasTest\Diagnostics\ChecksTest::testRedis
Redis tests are not enabled; enable them in phpunit.xml

/github/workspace/test/ChecksTest.php:117

3) LaminasTest\Diagnostics\ChecksTest::testMemcache
Memcache tests are not enabled; enable them in phpunit.xml

/github/workspace/test/ChecksTest.php:135

4) LaminasTest\Diagnostics\ChecksTest::testMemcached
Memcached tests are not enabled; enable them in phpunit.xml

/github/workspace/test/ChecksTest.php:152

5) LaminasTest\Diagnostics\ChecksTest::testMongo
Mongo tests are not enabled; enable them in phpunit.xml
```

Xerkus commented:
That got left behind when moving to current common CI setup I would guess and noone paid enough attention to notice.

Ocramius commented on src/Check/IniFile.php:
```diff
@@ -22,7 +22,7 @@ class IniFile extends AbstractFileCheck
     protected function validateFile($file)
     {
-        if (! is_array($ini = parse_ini_file($file)) || count($ini) < 1) {
+        if (! is_array($ini = @parse_ini_file($file)) || count($ini) < 1) {
```
Could we use something like `set_error_handler()` + `restore_error_handler()` here, perhaps? :thinking: 

Using `@` always bites us back later, so just wondering

heiglandreas replied:
I was indeed thinking about that and decided to skip it here for sake of brevity. But using the error handler might actually allow us to add the reported message to the error-message.

So will add

heiglandreas commented:
OK. Running the actual tests with the external tools fails. But it at least fails in all tested PHP-versions, so it's not related to PHP8.3

I created an issue to reintroduce these tests at #75

heiglandreas commented:
So shall I hit the "Merge" Button? Or is there anything missing?

heiglandreas committed:
Update .laminas-ci.json

Remove 'ignore-platform-reqs'

Signed-off-by: Andreas Heigl <andreas@heigl.org>

Chris53897 commented:
Thanks. Is there a release date planned?
